vignettes/sliding.Rmd

```diff
@@ -0,0 +1,281 @@
+
+Now we plot them on top of the latest case rates to see how the predictions compare.
+
+```{r plot-arx, message = FALSE, warning = FALSE, fig.width = 9, fig.height = 6}
+ggplot(fc, aes(x = fc_target_date, group = time_value, fill = engine_type)) + 
+  geom_line(data = x_latest, aes(x = time_value, y = case_rate),
+            inherit.aes = FALSE, color = "gray50") +
+  geom_ribbon(aes(ymin = `0.05`, ymax = `0.95`), alpha = 0.4) +
+  geom_line(aes(y = fc_.pred)) + geom_point(aes(y = fc_.pred), size = 0.5) +
+  geom_vline(aes(xintercept = time_value), linetype = 2, alpha = 0.5) +
+  facet_grid(vars(geo_value), vars(engine_type), scales = "free") +
+  scale_x_date(minor_breaks = "month", date_labels = "%b %y") +
+  labs(x = "Date", y = "Reported COVID-19 case rates") +
+  theme(legend.position = "none")
+```
+
+Simple linear regression clearly performs better than random forest in terms of accuracy of the predictions and not does not result in such in overconfident predictions (too narrow confidence bands). Though, in general, both approaches do not perform great. This is likely because we've only used two locations, and their behaviour is rather different from each other. 
+
+### Example using case data from Canada
+
+By leveraging the flexibility of `epiprocess`, we can apply the same techniques to data from other sources. Since some collaborators are in British Columbia, Canada, we'll do the same thing for Canada.
+
+The [COVID-19 Canada Open Data Working Group](https://opencovid.ca/) collects daily time series data on COVID-19 cases, deaths, recoveries, testing and vaccinations at the health region and province levels. Data are collected from publicly available sources such as government datasets and news releases. Unfortunately, there is no simple versioned source, so we have created our own from the Commit history.
+
+First, we load versioned case rates at the provincial level. After converting these to 7-day averages (due to highly variable provincial reporting mismatches), we then convert the data to an `epi_archive` object, and extract the latest version from it. Finally, we run a very similar forcasting exercise as that above, but here we compare producing the forcasts by using simple linear regression with using boosted regression trees.
+
+```{r get-can-fc, warning = FALSE}
+# source("drafts/canada-case-rates.R)
+can <- readRDS(
+  system.file("extdata", "can_prov_cases.rds", 
+              package = "epipredict", mustWork = TRUE)
+  ) %>%
+  group_by(version, geo_value) %>% 
+  arrange(time_value) %>% 
+  mutate(cr_7dav = RcppRoll::roll_meanr(case_rate, n = 7L)) #%>%
+  #filter(geo_value %in% c('Alberta', "BC"))
+can <- as_epi_archive(can)
+can_latest <- epix_as_of(can, max_version = max(can$DT$version))
+
+can_k_week_ahead <- function(ahead = 7, engine_type = "linear_reg") {
```

Comment:
Is there anyway to merge this function with the other one? Create 1 single function that can be used in both cases?